src/common/types.cpp

```diff
@@ -722,7 +722,11 @@ LogicalType LogicalType::MaxLogicalType(const LogicalType &left, const LogicalTy
 			child_types.emplace_back(left_child_types[i].first, std::move(child_type));
 		}
 
-		return LogicalType::STRUCT(child_types);
+		auto &return_aux = *left.GetAuxInfoShrPtr();
```

Comment:
Can we add `StructType::IsUnnamed` instead to encapsulate this logic and reduce boilerplate?